ts/features/bonus/cgn/components/merchants/CgnMerchantsFilters.tsx

```diff
@@ -0,0 +1,201 @@
+  };
+
+  const onConfirmButton = () => {
+    props.onConfirm();
+  };
+
+  const selectedFilters =
+    checkedCategories.length + (searchValue.length > 0 ? 1 : 0);
+
+  return (
+    <Container>
+      <AppHeader>
+        <Left />
+        <Body style={{ alignItems: "center" }}>
+          <H5 weight={"SemiBold"} color={"bluegrey"}>
+            {I18n.t("bonus.cgn.merchantsList.filter.title")}
+          </H5>
+        </Body>
+        <Right>
+          <ButtonDefaultOpacity onPress={props.onClose} transparent={true}>
+            <IconFont name="io-close" />
+          </ButtonDefaultOpacity>
+        </Right>
+      </AppHeader>
+      <SafeAreaView style={IOStyles.flex}>
+        <ScrollView style={[IOStyles.flex]} bounces={false}>
+          <View style={IOStyles.horizontalContentPadding}>
+            <H2>{I18n.t("bonus.cgn.merchantsList.filter.searchTitle")}</H2>
+            <Item>
+              <Input
+                value={searchValue}
+                autoFocus={true}
+                onChangeText={setSearchValue}
+                placeholderTextColor={IOColors.bluegreyLight}
+                placeholder={I18n.t(
+                  "bonus.cgn.merchantsList.filter.inputPlaceholder"
+                )}
+              />
+              <IconFont name="io-search" color={IOColors.bluegrey} />
+            </Item>
+            <View spacer large />
+            {props.isLocal && (
+              <>
+                <H2>{I18n.t("bonus.cgn.merchantsList.filter.addressTitle")}</H2>
+                <Item>
+                  <Input
+                    value={address}
+                    autoFocus={true}
+                    onChangeText={setAddress}
+                    placeholderTextColor={IOColors.bluegreyLight}
+                    placeholder={I18n.t(
+                      "bonus.cgn.merchantsList.filter.addressPlaceholder"
+                    )}
+                  />
+                  <IconFont name="io-place" color={IOColors.bluegrey} />
+                </Item>
+                <View spacer />
+                <DistanceSlider
+                  value={distance}
+                  onValueChange={setDistance}
+                  disabled={address.length === 0}
+                />
+                <View spacer large />
+              </>
+            )}
+            <H2>{I18n.t("bonus.cgn.merchantsList.filter.categories")}</H2>
+            <View spacer small />
+            <FlatList
+              data={categories}
+              keyExtractor={cat => cat.type}
+              keyboardShouldPersistTaps={"handled"}
+              ItemSeparatorComponent={() => (
+                <ItemSeparatorComponent noPadded={true} />
+              )}
+              renderItem={(listItem: ListRenderItemInfo<Category>) => (
+                <CategoryCheckbox
+                  text={I18n.t(listItem.item.nameKey)}
+                  value={listItem.item.type}
+                  checked={checkedCategories.indexOf(listItem.item.type) !== -1}
```

Comment:
```suggestion
                  checked={checkedCategories.some(listItem.item.type)}
```
same but less verbose

Reply:
Just find out that `some` is asking for a predicate so we have 2 options:
- checkedCategories.some(c => c === listItem.item.type)
- checkedCategories.includes(listItem.item.type)

I guess `includes` would be fine and less verbose